cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -137,4 +145,14 @@ public void run(TxRunnable runnable) throws TransactionException {
       throw new IllegalArgumentException(String.format("Namespace '%s' does not exist", namespace), nfe);
     }
   }
+
+  @Override
+  public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception {
+    String systemAppClassName = SystemAppTask.class.getName();
+    String systemAppParam = GSON.toJson(runnableTaskRequest);
+    RunnableTaskRequest taskRequest = RunnableTaskRequest.getBuilder(systemAppClassName)
```

Comment:
Why we need to build the runnable task request again? What's the difference between this one and the one from the argument?

Reply:
This one uses SystemAppTask as the task name. SystemAppTask will be a wrapper around the original task and do the classloading and context required to instantiate the original task.